### PR TITLE
fix: links should open a webview when not assigned to a native screen

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves []

### Description

trying to open https://www.artsy.net/art-appraisals was going to a 404 in the app. after this, it will open a webview.

we also need to put that link to the blocklist so it doesn't open the app, but its a good default to not go to 404 when we have forgotten to block the next one.

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Components/ArtsyReactWebView.tsx [hunk 1]
```diff
@@ -1,361 +0,0 @@
-import { OwnerType } from "@artsy/cohesion"
```
Comment (pvinis) [inline src/app/Components/ArtsyReactWebView.tsx:1]: i moved this file to `ArtsyWebView.*`

File: src/app/Scenes/SavedSearchAlert/screens/EmailPreferencesScreen.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const EmailPreferencesScreen: React.FC<Props> = (props) => {
   return (
     <Box flex={1}>
       <FancyModalHeader hideBottomDivider onLeftButtonPress={handleLeftButtonPress} />
-      <ArtsyReactWebView url="/unsubscribe" />
```
Comment (pvinis) [inline src/app/Scenes/SavedSearchAlert/screens/EmailPreferencesScreen.tsx:19]: `ArtsyReactWebView` was the webview, and `ArtsyWebView` was the webview with a fancymodal and a header.

Now `ArtsyWebView` is the webview, and `ArtsyWebViewPage` is the one with the header.

File: src/app/Scenes/VanityURL/VanityURLEntity.tsx [hunk 3]
```diff
@@ -54,6 +55,10 @@ interface RendererProps {
 }
 
 export const VanityURLEntityRenderer: React.FC<RendererProps> = ({ entity, slugType, slug }) => {
+  if (slugType === undefined) {
```
Comment (pvinis) [inline src/app/Scenes/VanityURL/VanityURLEntity.tsx:58]: main change here

File: src/app/Components/ArtsyWebView.tsx [hunk 4]
```diff
@@ -1,6 +1,361 @@
...
+import { Platform } from "react-native"
+import Share from "react-native-share"
+import WebView, { WebViewProps } from "react-native-webview"
+import { useTracking } from "react-tracking"
+import { useScreenDimensions } from "shared/hooks"
+import { ArtsyKeyboardAvoidingView } from "shared/utils"
+import { parse as parseURL } from "url"
+import { FancyModalHeader } from "./FancyModal/FancyModalHeader"
 
-export const ArtsyWebView: React.FC<{ url: string }> = ({ url }) => {
```
Comment (pvinis) [inline src/app/Components/ArtsyWebView.tsx:4]: this file was just a wrapper. i moved the whole thing here.

File: src/app/Components/ArtsyWebView.tests.tsx [hunk 5]
```diff
@@ -1,12 +1,306 @@
...
+}))
 
 describe("ArtsyWebView", () => {
   it("shows react-native-webview", () => {
     const tree = renderWithWrappers(<ArtsyWebView url="random-url" />)
     expect(tree.root.findAllByType(WebView).length).toEqual(1)
   })
 })
+
+describe("ArtsyWebViewPage", () => {
```
Comment (pvinis) [inline src/app/Components/ArtsyWebView.tests.tsx:42]: keeping the one test we had, plus adding all the ones from the wrapped component.

**Review by brainbicycle — APPROVED** [review]
Thanks for the fix! 
nitpick (non-blocking): I think the cleanup code could have been a separate PR?

**Comment by pvinis** [comment]
>  nitpick (non-blocking): I think the cleanup code could have been a separate PR?
its true.. shall it? i should I guess

**Comment by pvinis** [comment]
done

**Comment by brainbicycle** [comment]
I made a little modification to the behavior here, I think this broke when we introduced the custom not found view for 404 errors. What should happen in the case of a bare url like this: we try to render VanityURL component, this first tries to see if there is a profile associated with the url (partner or gallery), if there isn't (404) we render the VanityURLRedirect component which will see if there is a force redirect for that url, if there is a redirect we render whatever component that redirect points to, if there isn't we open the url in a webview. Since the error that is supposed to show the VanityURLRedirect is a 404 this was not happening and instead we were showing the custom 404 view. I added a param to opt out of this behavior just for this component, I think for most components it is what we want this is just a bit of a special case.

**Review by olerichter00 — APPROVED** [review]
Thanks for the fix 🙇

**Comment by brainbicycle** [comment]
here are the two problem links working:

https://user-images.githubusercontent.com/49686530/173826342-f6f1dbfd-26ab-48ff-92de-03305d005096.mov

**Pull request merged**